api/ocm/compdesc/meta/v1/identity.go

```diff
@@ -103,6 +103,22 @@ func (i Identity) Remove(name string) bool {
 	return false
 }
 
+// ExtraIdentity extracts the extra identity part of an identity.
+func (i Identity) ExtraIdentity() Identity {
+	if i == nil {
+		return nil
+	}
+	if _, ok := i[SystemIdentityName]; !ok {
+		return i
+	}
+	if len(i) == 1 {
+		return nil
+	}
```

Comment:
Can you explain the logic here please?

If there is no system identity, then just returning the identity is a false positive, because that's not the ExtraIdentity that's just the identity, isn't it? Does this mean that if there is no system identity, the rest is just extra identity?

So anything else that isn't the system identity, is the extra identity, but why is it a problem if it's just one thing?

Reply:
The identity type may represent any set of identity attributes, not necessarily a complete identity.
The method ExtraIdentity just filters the attributes, which would be part of the extra identity of a complete element identity. Meaning: remove the SystemIdentityName attribute if set. If only this attribute is set, it would have no extra identity.

Reply:
Okay, but if the systemIdentity is a part of the extra identity, why is it filtered out? And if it IS filtered out, why is it part of the extra identity? 🤔

Reply:
The extra identity is the content of the field `extraIdentity` plus optionally the `version``, but it never includes the `name` attribute. In the future we will require the `version`to be explicitly added to the `extraIdentity`field if it should be used, instead of implicitly adding if required to be unique.

So, formally, the identity of an element is the `name`of the element plut the attributed described by the `extraIdentity` field.  And vice versa, the extra identity of an element is the identity without the element name.